### PR TITLE
:seedling: Update vcsim tests under WCP_Workload_Domain_Isolation

### PR DESCRIPTION
**Pull request by zyiyi11**
**What does this PR do, and why is it needed?**
This patch updates vcsim tests when WCP_Workload_Domain_Isolation enabled. 
- Updated `CreateWorkloadNamespace` to create namespaced Zone CR when FSS enabled.
- Test both FSS enabled/disabled for `zone_placement_test.go`.
- Test only FSS enabled for `folder_test.go`, `vmprovider_resourcepolicy_test.go` ect because they only cares `nsInfo.Namespace` during tests.

**Which issue(s) is/are addressed by this PR?** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Fixes # N/A

**Are there any special notes for your reviewer**:

**Please add a release note if necessary**:

```release-note
NONE
```

### CODE REVIEW

File: test/builder/vcsim_test_context.go [hunk 1]
```diff
@@ -249,14 +252,31 @@ func (c *TestContextForVCSim) CreateWorkloadNamespace() WorkloadNamespaceInfo {
...
+		// When FSS_WCP_WORKLOAD_DOMAIN_ISOLATION is disabled, AvailabilityZone stores namespace info.
+		if !config.WithWorkloadIsolation {
+			az := &topologyv1.AvailabilityZone{}
+			Expect(c.Client.Get(c, client.ObjectKey{Name: azName}, az)).To(Succeed())
+			if az.Spec.Namespaces == nil {
+				az.Spec.Namespaces = map[string]topologyv1.NamespaceInfo{}
+			}
+			az.Spec.Namespaces[ns.Name] = nsInfo
+			Expect(c.Client.Update(c, az)).To(Succeed())
+			// When FSS_WCP_WORKLOAD_DOMAIN_ISOLATION is enabled, Namespaced Zone stores namespace info.
```
Comment (bryanv) [inline test/builder/vcsim_test_context.go:264]: It is probably clearer to move this comment after the `} else {`

File: test/builder/vcsim_test_context.go [hunk 2]
```diff
@@ -218,7 +221,7 @@ func (c *TestContextForVCSim) AfterEach() {
 	c.UnitTestContext.AfterEach()
 }
 
-func (c *TestContextForVCSim) CreateWorkloadNamespace() WorkloadNamespaceInfo {
+func (c *TestContextForVCSim) CreateWorkloadNamespace(config VCSimTestConfig) WorkloadNamespaceInfo {
```
Comment (bryanv) [inline test/builder/vcsim_test_context.go:224]: Save the value in the `TestContextForVCSim` so this doesn't need to be updated.

Comment (zyiyi11) [reply test/builder/vcsim_test_context.go]: I am going with `save the value in the TestContextForVCSim`

File: test/builder/vcsim_test_context.go [hunk 3]
```diff
@@ -144,6 +147,9 @@ type TestContextForVCSim struct {
 	datastore *object.Datastore
 
 	azCCRs map[string][]*object.ClusterComputeResource
+
+	// WithWorkloadIsolation stores VCSimTestConfig.WithWorkloadIsolation value.
```
Comment (bryanv) [inline test/builder/vcsim_test_context.go:151]: I don't think this needs to be public. Also group this near the existing `WithFaultDomains ...` comment (and maybe update that comment a bit).

**Review by bryanv — APPROVED** [review]

**Pull request merged**